great_expectations/validator/v1_validator.py

```diff
@@ -0,0 +1,92 @@
+from __future__ import annotations
+
+from enum import Enum
+from typing import TYPE_CHECKING
+
+from great_expectations.core.expectation_validation_result import (
+    ExpectationSuiteValidationResult,
+    ExpectationValidationResult,
+)
+from great_expectations.validator.validator import calc_validation_statistics
+
+if TYPE_CHECKING:
+    from great_expectations.core import ExpectationSuite
+    from great_expectations.core.batch_config import BatchConfig
+    from great_expectations.data_context import AbstractDataContext
+    from great_expectations.expectations.expectation import (
+        Expectation,
+        ExpectationConfiguration,
+    )
+
+
+class ResultFormat(str, Enum):
+    BOOLEAN_ONLY = "BOOLEAN_ONLY"
+    BASIC = "BASIC"
+    SUMMARY = "SUMMARY"
+    COMPLETE = "COMPLETE"
+
+
+class Validator:
+    """Validator.
+
+    Responsible for running expectations on a batch configuration.
+    """
+
+    context: AbstractDataContext
+    batch_config: BatchConfig
+    batch_asset_options: dict | None
+    result_format: ResultFormat
+
+    def __init__(
+        self,
+        context: AbstractDataContext,
+        batch_config: BatchConfig,
+        result_format: ResultFormat = ResultFormat.SUMMARY,
+        batch_asset_options: dict | None = None,
```

Comment:
Nit - where should context come in terms of ordering? I think we should be consistent across our API or require mandatory key-word args to improve UX

Reply:
Seems reasonable-ish. I don't have strong opinions, but in any code I've written, I've tended to think in terms of putting the most "core" entities first. Nothing makes sense without context, so I put that first. But I'm fine to go with whatever. I love mandating keyword args for internal use. Do we like forcing that on oss users?

Reply:
Just mainly thinking out loud - I also don't care too much here.

I agree with probably not forcing keyword args for users let's leave what you have.